Index: setup/install_desktop_packages.sh
===================================================================
--- ./astrobee/src/astrobee/scripts/setup/install_desktop_packages
+++ ./astrobee/src/astrobee/scripts/setup/install_desktop_packages.sh.new
@@ -50,27 +50,27 @@ then
  
   if [ "${NO_TUNNEL}" -eq 1 ]; then
       echo "Getting the custom Debian without tunnel"
-      sudo /bin/bash -c "echo \"deb [arch=amd64] http://astrobee.ndc.nasa.gov/software ${DIST} main\" > $arssrc" || exit 1
-      sudo /bin/bash -c "echo \"deb-src http://astrobee.ndc.nasa.gov/software ${DIST} main\" >> $arssrc" || exit 1
+      sudo -E /bin/bash -c "echo \"deb [arch=amd64] http://astrobee.ndc.nasa.gov/software ${DIST} main\" > $arssrc" || exit 1
+      sudo -E /bin/bash -c "echo \"deb-src http://astrobee.ndc.nasa.gov/software ${DIST} main\" >> $arssrc" || exit 1
   else
       echo "Tunnelling to get the custom Debian"
-      sudo /bin/bash -c "echo \"deb [arch=amd64] http://127.0.0.1:8765/software ${DIST} main\" > $arssrc" || exit 1
-      sudo /bin/bash -c "echo \"deb-src http://127.0.0.1:8765/software ${DIST} main\" >> $arssrc" || exit 1
+      sudo -E /bin/bash -c "echo \"deb [arch=amd64] http://127.0.0.1:8765/software ${DIST} main\" > $arssrc" || exit 1
+      sudo -E /bin/bash -c "echo \"deb-src http://127.0.0.1:8765/software ${DIST} main\" >> $arssrc" || exit 1
       ssh -N -L 8765:astrobee.ndc.nasa.gov:80 ${username}m.ndc.nasa.gov &
   fi
   
-  trap "kill $! 2> /dev/null; sudo truncate -s 0 $arssrc; wait $!" 0 HUP QUIT ILL ABRT FPE SEGV PIPE TERM INT
+  trap "kill $! 2> /dev/null; sudo -E truncate -s 0 $arssrc; wait $!" 0 HUP QUIT ILL ABRT FPE SEGV PIPE TERM INT
   sleep 1
 fi
 
-sudo apt-get update || exit 1
+sudo -E apt-get update || exit 1
 
-if ! sudo apt-get install -m -y $pkgs; then
+if ! sudo -E apt-get install -m -y $pkgs; then
   filter_pkgs="libroyale-dev rti-dev libsoracore-dev libmiro-dev libroyale1 rti libmiro0 libsoracore1"
   for p in $filter_pkgs; do
     pkgs=${pkgs//$p/}
   done
-  sudo apt-get install -m -y $pkgs || {
+  sudo -E apt-get install -m -y $pkgs || {
     echo "Couldn't install a necessary package."
     exit 1
   }
