Index: opencv/cmake/OpenCVDownload.cmake
===================================================================
--- opencv.orig/cmake/OpenCVDownload.cmake
+++ opencv/cmake/OpenCVDownload.cmake
@@ -13,6 +13,8 @@
 #    UNPACK - downloaded file will be unpacked to DESTINATION_DIR
 #    RELATIVE_URL - if set, then URL is treated as a base, and FILENAME will be appended to it
 #  Note: uses OPENCV_DOWNLOAD_PATH folder as cache, default is <opencv>/.cache
+set(ENV{HTTP_PROXY} "")
+set(ENV{HTTPS_PROXY} "")
 
 set(HELP_OPENCV_DOWNLOAD_PATH "Cache directory for downloaded files")
 if(DEFINED ENV{OPENCV_DOWNLOAD_PATH})
