--- ./src/astrobee/description/description/urdf/sensor_nav_cam.urdf.xacro	2021-03-22 18:12:44.070400245 +0900
+++ ./src/astrobee/description/description/urdf/sensor_nav_cam.urdf.xacro.new	2021-03-22 18:12:44.070400245 +0900
@@ -24,10 +24,10 @@
       <always_on>true</always_on>
       <update_rate>0.0001</update_rate>
       <camera>
-        <horizontal_fov>2.957216683909311</horizontal_fov>
+        <horizontal_fov>2.21376499</horizontal_fov>
         <image>
-          <width>320</width>
-          <height>240</height>
+          <width>1280</width>
+          <height>960</height>
           <format>L8</format>
         </image>
         <clip>
@@ -35,9 +35,16 @@
           <far>20</far>
         </clip>
         <distortion>
-          <k1>0.993308</k1>
+          <k1>-0.312191</k1>
+          <k2>0.073843</k2>
+          <p1>-0.000918</p1>
+          <p2>0.001890</p2>
         </distortion>
+        <lens>
+          <env_texture_size>1024</env_texture_size>
+        </lens>
       </camera>
+
       <plugin name="nav_cam_ros" filename="libgazebo_sensor_plugin_nav_cam.so">
         <namespace>/${ns}/</namespace>
       </plugin>
